extensions/azurePublish/src/components/ResourceGroupPicker.tsx

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as React from 'react';
+import formatMessage from 'format-message';
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+import { Icon } from 'office-ui-fabric-react/lib/Icon';
+import { Stack } from 'office-ui-fabric-react/lib/Stack';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import { FluentTheme } from '@uifabric/fluent-theme';
+import { TooltipHost } from 'office-ui-fabric-react/lib/Tooltip';
+
+import { useDebounce } from './useDebounce';
+
+const stackStyles = { root: { marginBottom: '8px' } };
+const dropdownStyles = { root: { marginBottom: '8px' }, dropdown: { width: '100%' } };
+const itemIconStyles = { marginRight: '8px' };
+const newNameTextFileStyles = { root: { marginTop: '10px' } };
+
+const getInfoIconStyle = (required) => {
+  return {
+    root: {
+      selectors: {
+        '&::before': {
+          content: required ? " '*'" : '',
+          color: FluentTheme.palette.red,
+          paddingRight: 3,
+        },
+      },
+    },
+  };
+};
+
+const CREATE_NEW_KEY = 'CREATE_NEW';
+
+const createNewOption: IDropdownOption = {
+  key: CREATE_NEW_KEY,
+  data: { icon: 'Add' },
+  text: formatMessage('Create new'),
+};
+
+type ResourceGroupItemChoice = {
+  name: string;
+  isNew: boolean;
+  errorMessage?: string;
+};
+
+type Props = {
+  /**
+   * The resource groups to choose from.
+   * Set to undefined to disable this picker.
+   */
+  resourceGroupNames?: string[];
+  /**
+   * The selected name of the existing resource.
+   * When undefined, the 'Create new' option will be selected.
+   */
+  selectedName?: string;
```

Comment:
selectedGroupName

Reply:
Ha! I shortened the names from these just before check-in thinking that it make it more readable.  You are right that the longer names are better as they describe the contract.